example/airtable/elden-ring-map/register.php

```diff
@@ -15,21 +15,35 @@
 function register_airtable_elden_ring_map_block() {
 	$block_name   = 'Elden Ring Location';
 	$access_token = \RemoteDataBlocks\Example\get_access_token( 'airtable_elden_ring' );
-	$base         = 'appqI3sJ9R2NcML8Y';
-	$tables       = [
-		'maps'      => 'tblS3OYo8tZOg04CP',
-		'locations' => 'tblc82R9msH4Yh6ZX',
-	];
 
 	if ( empty( $access_token ) ) {
 		$logger = LoggerManager::instance();
 		$logger->warning( sprintf( '%s is not defined, cannot register %s block', 'EXAMPLE_AIRTABLE_ELDEN_RING_ACCESS_TOKEN', $block_name ) );
 		return;
 	}
 
-	$elden_ring_datasource = new AirtableDatasource( $access_token, $base, $tables );
-	$list_locations_query  = new AirtableEldenRingListLocationsQuery( $elden_ring_datasource );
-	$list_maps_query       = new AirtableEldenRingListMapsQuery( $elden_ring_datasource );
+	$locations_config = [
+		'friendly_name'   => $block_name . ' (Locations)',
+		'uid'             => 'appqI3sJ9R2NcML8Y/tblc82R9msH4Yh6ZX',
+		'endpoint'        => 'https://api.airtable.com/v0/appqI3sJ9R2NcML8Y/tblc82R9msH4Yh6ZX',
+		'request_headers' => [
+			'Authorization' => "Bearer {$access_token}",
+			'Content-Type'  => 'application/json',
+		],
+	];
+
+	$maps_config = [
+		'friendly_name'   => $block_name . ' (Maps)',
+		'uid'             => 'appqI3sJ9R2NcML8Y/tblS3OYo8tZOg04CP',
+		'endpoint'        => 'https://api.airtable.com/v0/appqI3sJ9R2NcML8Y/tblS3OYo8tZOg04CP',
+		'request_headers' => [
```

Comment:
I like the config approach. But our agreement was that v1 would be code-based and, after learning from additional use cases, we would tackle a more static config approach. I worry that we don't have enough examples under our belt to understand what limitations this brings, and we will bifurcate the happy path.

Reply:
FWIW, on another pass, I would take it this direction and probably stop here.

_Not_ pure json.

In the quest for data isolation from logic, I forgot about the perf implications. It's rough in PHP to spend so much time in json (or yaml) file loading on every request. So, associative arrays, in PHP, to leverage opcache. And we retain that executable flexibility.